core/src/test/java/org/apache/iceberg/TestDeleteFiles.java

```diff
@@ -68,38 +68,46 @@ public class TestDeleteFiles extends TableTestBase {
                   ))
           .build();
 
-  @Parameterized.Parameters(name = "formatVersion = {0}")
+  private final String branch;
+
+  @Parameterized.Parameters(name = "formatVersion = {0}, branch = {1}")
   public static Object[] parameters() {
-    return new Object[] {1, 2};
+    return new Object[][] {
+      new Object[] {1, "main"},
+      new Object[] {1, "testBranch"},
+      new Object[] {2, "main"},
+      new Object[] {2, "testBranch"}
+    };
   }
 
-  public TestDeleteFiles(int formatVersion) {
+  public TestDeleteFiles(int formatVersion, String branch) {
     super(formatVersion);
+    this.branch = branch;
   }
 
   @Test
   public void testMultipleDeletes() {
-    table.newAppend().appendFile(FILE_A).appendFile(FILE_B).appendFile(FILE_C).commit();
+    Snapshot append =
+        commit(
+            table,
+            table.newAppend().appendFile(FILE_A).appendFile(FILE_B).appendFile(FILE_C),
+            branch);
 
     Assert.assertEquals("Metadata should be at version 1", 1L, (long) version());
-    Snapshot append = readMetadata().currentSnapshot();
```

Comment:
This is actually different validation behavior because the `readMetadata()` method doesn't go through the table. It goes directly to the backing store to get the metadata. I'm not sure that this is a problem, but I'd probably lean toward keeping the existing behavior. We can just make a version of `latestSnapshot` that accepts `TableMetadata` as returned by `readMetadata()`.

Reply:
I've updated with an additional latestSnapshot method which takes in `TableMetadata` . The cases which were using it before, are now using this method instead of through the table.